### PR TITLE
HYC-1490 - Test coverage/refactor for attach_files_to_work_job overrides

### PR DESCRIPTION
**Pull request by bbpennel**
https://jira.lib.unc.edu/browse/HYC-1490

* Add tests for overrides
* Change approach to override, removing non-overridden code, and wrapping the original method
* Change infected file behavior, so it now will only delete the pattern dir containing the file if its empty. Also, catch any errors that occur during deletion so email always gets sent.

### CODE REVIEW

File: spec/jobs/attach_files_to_work_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      expect(File).not_to exist(target_file)
+    end
+
+    context 'that cannot be deleted' do
+      before do
+        allow(File).to receive(:delete).and_raise(Errno::EACCES)
+      end
+
+      it 'still sends virus email message' do
+        subject.perform(work, [uploaded_file])
```
Comment (lfarrell) [inline spec/jobs/attach_files_to_work_job_spec.rb:49]: There's no expect, just the action that needs to be checked.

File: spec/jobs/attach_files_to_work_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      end
+    end
+  end
+
+  context 'with safe file' do
```
Comment (lfarrell) [inline spec/jobs/attach_files_to_work_job_spec.rb:54]: Shouldn't this end with an expect as well?

File: spec/jobs/attach_files_to_work_job_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      # Original method should not be called
+      allow(subject).to receive(:original_perform).and_raise('nope')
+      allow(Hyc::VirusScanner).to receive(:hyc_infected?).and_return(ClamAV::VirusResponse.new(target_file, 'avirus'))
+      expect(Hyrax::Workflow::VirusFoundNotification).to receive(:send_notification)
+          .with(entity: entity, recipients: anything, comment: anything, user: user)
+    end
+
+    it 'sends virus email message' do
+      subject.perform(work, [uploaded_file])
+      expect(File).not_to exist(target_file)
```
Comment (lfarrell) [inline spec/jobs/attach_files_to_work_job_spec.rb:40]: Seems like this isn't testing whether an email was sent.

**Review by lfarrell — CHANGES_REQUESTED** [review]
Added a few comments, though I may just be misunderstanding what the tests are doing.

File: spec/jobs/attach_files_to_work_job_spec.rb (hunk 3 above)
Comment (bbpennel) [reply spec/jobs/attach_files_to_work_job_spec.rb:40]: the expect is up in the `before` section. I could move it into the `it`, but would have to duplicate it in the next test as well. Do you think that would be better for clarity?

File: spec/jobs/attach_files_to_work_job_spec.rb (hunk 2 above)
Comment (bbpennel) [reply spec/jobs/attach_files_to_work_job_spec.rb:54]: When verifying that a method is called, you have to setup the `expect` before executing the code that would call it

File: spec/jobs/attach_files_to_work_job_spec.rb (hunk 3 above)
Comment (lfarrell) [reply spec/jobs/attach_files_to_work_job_spec.rb:40]: Maybe a comment.

**Pull request merged**